catgrad/src/interpreter/types.rs

```diff
@@ -12,7 +12,7 @@ pub type ResultValues<B> = abstract_interpreter::ResultValues<Interpreter<B>>;
 // Multiple tagged ndarrays
 
 // TODO: make this sealed
-pub trait HasDtype: Copy + Send + Sync + std::fmt::Debug {}
+pub trait HasDtype: Copy + Send + Sync + std::fmt::Debug + 'static {}
```

Comment:
Shouldn't we be able to fully delete HasDtype now?

Reply:
the ndarray backend still uses generic functions in HasDtype so that remains to be done after these changes.